exetera/core/field_importers.py

```diff
@@ -0,0 +1,344 @@
+from typing import Mapping
+from exetera.core.abstract_types import DataFrame
+import numpy as np
+from exetera.core import fields as fld
+from exetera.core import operations as ops
+from exetera.core.data_writer import DataWriter
+from exetera.core import utils
+from datetime import datetime, date
+
+
+FIELD_MAPPING_TO_IMPORTER = {
+    'categorical': lambda categories, value_type, allow_freetext:
+                   lambda s, df, name, ts: LeakyCategoricalImporter(s, df, name, categories, value_type, ts) if allow_freetext else CategoricalImporter(s, df, name, categories, value_type, ts),
+    'numeric': lambda dtype, invalid_value, validation_mode, create_flag_field, flag_field_name:
+               lambda s, df, name, ts: NumericImporter(s, df, name, dtype, invalid_value, validation_mode, create_flag_field, flag_field_name, timestamp=ts),
+    'string': lambda fixed_length:
+              lambda s, df, name, ts: IndexedStringImporter(s, df, name, ts) if fixed_length is None else FixedStringImporter(s, df, name, fixed_length, ts),
+    'datetime': lambda create_day_field, create_flag_field:
+                lambda s, df, name, ts: DateTimeImporter(s, df, name, create_day_field, create_flag_field, ts),
+    'date': lambda create_flag_field:
+            lambda s, df, name, ts: DateImporter(s, df, name, create_flag_field, ts),
+}
+
+#========= ImporterDefinition, include Categorical, Numeric, , , Datetime =========
+
+class ImporterDefinition:
+    def __init__(self):
+        self._field_size = 0
+        self._importer = None
+
+
+class Categorical(ImporterDefinition):
```

Comment:
All these importer definition derived types should be documented; they are part of the API

Reply:
I think it would be better to name field_importers.py to just be importers.py. Once reader importers are gone they will be the only importer types

Reply:
We already have a script named "importer.py" for the importer command line though.